File: src/Sequence/world/meta/Formula.java
package Sequence.world.meta;

import Sequence.world.meta.imagine.ImagineEnergyRecord;
import mindustry.type.ItemStack;
import mindustry.type.LiquidStack;

public class Formula {
    public final ItemStack[] inputItem;
    public final LiquidStack[] inputLiquid;
    public final float inputPower;
    public final ImagineEnergyRecord inputImagine;
    public final ItemStack[] outputItem;
    public final LiquidStack[] outputLiquid;
    public final float outputPower;
    public final ImagineEnergyRecord outputImagine;
    public final float time;
    public final boolean liquidSecond;

    public Formula(ItemStack[] inputItem, LiquidStack[] inputLiquid, float inputPower, ImagineEnergyRecord inputImagine, ItemStack[] outputItem, LiquidStack[] outputLiquid, float outputPower, ImagineEnergyRecord outputImagine, float time, boolean liquidSecond) {
        this.inputItem = inputItem;
        this.inputLiquid = inputLiquid;
        this.inputPower = inputPower;
        this.inputImagine = inputImagine;
        this.outputItem = outputItem;
        this.outputLiquid = outputLiquid;
        this.outputPower = outputPower;
        this.outputImagine = outputImagine;
        this.time = time;
        this.liquidSecond = liquidSecond;
    }

    public Formula(ItemStack[] inputItem, LiquidStack[] inputLiquid, float inputPower, ItemStack[] outputItem, LiquidStack[] outputLiquid, float outputPower, float time, boolean liquidSecond) {
        this(inputItem, inputLiquid, inputPower, ImagineEnergyRecord.empty(), outputItem, outputLiquid, outputPower, ImagineEnergyRecord.empty(), time, liquidSecond);
    }

    public Formula(ItemStack[] inputItem, LiquidStack[] inputLiquid, float inputPower, ItemStack[] outputItem, LiquidStack[] outputLiquid, float outputPower, float time) {
        this(inputItem, inputLiquid, inputPower, outputItem, outputLiquid, outputPower, time, false);
    }
}

File: src/sequence/world/meta/Formula.kt
package sequence.world.meta

import mindustry.type.ItemStack
import mindustry.type.LiquidStack
import sequence.world.meta.imagine.ImagineEnergyRecord

class Formula(
    val inputItem: Array<ItemStack>,
    val inputLiquid: Array<LiquidStack>,
    val inputPower: Float,
    val inputImagine: ImagineEnergyRecord,
    val outputItem: Array<ItemStack>,
    val outputLiquid: Array<LiquidStack>,
    val outputPower: Float,
    val outputImagine: ImagineEnergyRecord,
    val time: Float,
    val liquidSecond: Boolean
) {
    @JvmOverloads
    constructor(
        inputItem: Array<ItemStack>,
        inputLiquid: Array<LiquidStack>,
        inputPower: Float,
        outputItem: Array<ItemStack>,
        outputLiquid: Array<LiquidStack>,
        outputPower: Float,
        time: Float,
        liquidSecond: Boolean = false
    ) : this(
        inputItem,
        inputLiquid,
        inputPower,
        ImagineEnergyRecord.empty,
        outputItem,
        outputLiquid,
        outputPower,
        ImagineEnergyRecord.empty,
        time,
        liquidSecond
    )
}
